queue/queue.go

```diff
@@ -36,13 +36,14 @@ func (db *Store) Next(ctx context.Context, resp mdm.Response) (*Command, error)
 	var cmd *Command
 	switch resp.Status {
 	case "NotNow":
-		// move down, send next
+		// We will try this command later when the device is not
+		// responding with NotNow
 		x, a := cut(dc.Commands, resp.CommandUUID)
 		dc.Commands = a
 		if x == nil {
 			break
 		}
-		dc.Commands = append(dc.Commands, *x)
+		dc.NotNow = append(dc.NotNow, *x)
```

Comment:
What does the device see in the response? Just 200 OK with no response?

Have you seen the `Handling a NotNow Response` in the documentation?
https://developer.apple.com/library/content/documentation/Miscellaneous/Reference/MobileDeviceManagementProtocolRef/3-MDM_Protocol/MDM_Protocol.html#//apple_ref/doc/uid/TP40017387-CH3-SW406

Reply:
If the Command returned by Next is nil, the response will be empty yes.

And yes this solution depends on the device polling later basically.